akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/HttpChallenge.scala

```diff
@@ -11,9 +11,9 @@ import akka.http.impl.util.JavaMapping.Implicits._
 
 final case class HttpChallenge(scheme: String, realm: String,
                                params: Map[String, String] = Map.empty) extends jm.headers.HttpChallenge with ValueRenderable {
-
   def render[R <: Rendering](r: R): r.type = {
-    r ~~ scheme ~~ " realm=" ~~#! realm
+    r ~~ scheme
+    if (realm != null && !realm.isEmpty) r ~~ " realm=" ~~#! realm
```

Comment:
I considered that, but I could not see a legitimate reason for rendering
`WWW-Authenticate: Basic realm=""`
vs 
`WWW-Authenticate: Basic`
Having I can't see a reason *not* to permit it and perhaps as you suggest there are applications out there that really do expect realm to be present and an empty string. I'll permit it in the next push.